stdlib/ballerina-builtin/src/main/ballerina/ballerina/io/natives.bal

```diff
@@ -75,85 +94,64 @@ public native function createCharacterChannel (ByteChannel byteChannel, string e
 @Return {value:"Number of bytes written"}
-public native function <ByteChannel channel> writeBytes (blob content, int startOffset) (int);
+public native function <ByteChannel channel> write (blob content, int startOffset, int numberOfBytes) (int, IOError);
 
-@Description {value:"Function to close a byte channel"}
-@Param {value:"channel: The ByteChannel to be closed"}
-public native function <ByteChannel channel> close ();
+@Description {value:"Function to read characters"}
+@Param {value:"channel: The CharacterChannel to read characters from"}
+@Param {value:"numberOfChars: Number of characters which should be read"}
+@Return {value:"The character sequence which was read"}
+public native function <CharacterChannel channel> readCharacters (int numberOfChars) (string, IOError);
+
+@Description {value:"Function to write characters"}
+@Param {value:"channel: The CharacterChannel which will allow writing characters"}
+@Param {value:"content: Text content which should be written"}
+@Param {value:"startOffset: If the content needs to be written with an offset, the value of that offset"}
+@Return {value:"Number of characters written"}
+public native function <CharacterChannel channel> writeCharacters (string content, int startOffset) (int, IOError);
 
 @Description {value:"Function to check whether next record is available or not"}
 @Param {value:"channel: The DelimitedRecordChannel to read text records from"}
 @Return {value:"True if the channel has more records; false otherwise"}
 public native function <DelimitedRecordChannel channel> hasNextTextRecord () (boolean);
 
-@Description {value:"Function to return a ByteChannel related to the file. This ByteChannel can then be used to read/write from/to the file."}
-@Param {value:"path: The path of the file"}
-@Param {value:"accessMode: Specifies whether the file should be opened for reading or writing (r/w)"}
-@Return {value:"ByteChannel which will allow to perform I/O operations"}
-public native function openFile (string path, string accessMode) (ByteChannel);
+@Description {value:"Function to read text records"}
+@Param {value:"channel: The DelimitedRecordChannel to read text records from"}
+@Return {value:"Fields listed in the record"}
+public native function <DelimitedRecordChannel channel> nextTextRecord () (string[], IOError);
 
-@Description {value:"Represents a client socket connection. This can be used to communicate with a remote server"}
-@Param {value:"host: Remote server domain/IP"}
-@Param {value:"port: Remote server port"}
-@Param {value:"options: Connection stream that bridge the client and the server"}
-@Return {value:"Socket which will allow to communicate with a remote server"}
-public native function openSocket (string host, int port, SocketProperties options) (Socket);
+@Description {value:"Function to write text records"}
+@Param {value:"channel: The DelimitedRecordChannel to write records"}
+@Param {value:"records: Fields which are included in the record"}
```

Comment:
Document the returned error as well. Shall we check all the methods?

Reply:
Added the return with 3b58422f6c457c12c1ed16661b17ae49a52a17a7